### PR TITLE
New ivar_variants_to_vcf.py

### PR DESCRIPTION
**Pull request by Shettland**
## PR checklist

- [X] Refactored old ivar_variants_to_vcf.py with new functionalities like merging duplicate variants in the same position based on AF differencies and a new quality threshold.
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/viralrecon/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/viralrecon _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: bin/ivar_variants_to_vcf.py [hunk 1]
```diff
@@ -34,18 +33,38 @@ def parse_args(args=None):
         default=0,
         help="Only output variants where allele frequency is greater than this number (default: 0).",
     )
+    parser.add_argument(
+        "-bq",
+        "--bad_quality_threshold",
+        type=float,
```
Comment (saramonzon) [inline bin/ivar_variants_to_vcf.py]: type int

File: bin/ivar_variants_to_vcf.py [hunk 2]
```diff
@@ -57,537 +76,707 @@ def parse_args(args=None):
...
+                    + record.id
+                    + ",length="
+                    + str(len(record.seq))
+                    + ">"
+                ]
+
+            header_source += header_contig
+
+        header_info = [
+            '##INFO=<ID=TYPE,Number=1,Type=String,Description="Either SNP (Single Nucleotide Polymorphism), DEL (deletion) or INS (Insertion)">',
```
Comment (saramonzon) [inline bin/ivar_variants_to_vcf.py:694]: should this be Number=1? It's a string right?

Comment (Shettland) [reply bin/ivar_variants_to_vcf.py:694]: I'll update all the header, its not only that one

File: bin/ivar_variants_to_vcf.py [hunk 3]
```diff
@@ -57,537 +76,709 @@ def parse_args(args=None):
...
+                    cleaned_ref_rows_list.append(df)
+        final_ref_rows_list = remove_subsets(cleaned_ref_rows_list)
+        return final_ref_rows_list
+
+    def process_vcf_df(self, vcf_df):
+        """Merge rows with consecutive SNPs that passed all filters and without NAs
+
+        Args:
+            vcf_df - dataframe from self.initiate_vcf_df()
+            consensus - wether to merge only variants meeting consensus AF criteria
```
Comment (saramonzon) [inline bin/ivar_variants_to_vcf.py]: no consensus arg in this function

**Pull request merged**